src/core/server/saved_objects/migrationsv2/integration_tests/actions.test.ts

```diff
@@ -0,0 +1,449 @@
+            {
+              remove: {
+                alias: 'no_such_alias',
+                index: 'existing_index_1',
+                must_exist: true,
+              },
+            },
+          ]);
+          return expect(task()).resolves.toMatchInlineSnapshot(`
+                    Object {
+                      "_tag": "Left",
+                      "left": Object {
+                        "type": "alias_not_found_exception",
+                      },
+                    }
+                  `);
+        });
+      });
+    });
+    describe('remove_index', () => {
+      it('left index_not_found_exception if index does not exist', () => {
+        const task = updateAliases(client, [
+          {
+            remove_index: {
+              index: 'no_such_index',
+            },
+          },
+        ]);
+        return expect(task()).resolves.toMatchInlineSnapshot(`
+                  Object {
+                    "_tag": "Left",
+                    "left": Object {
+                      "index": "no_such_index",
+                      "type": "index_not_found_exception",
+                    },
+                  }
+                `);
+      });
+      it('left remove_index_not_a_concrete_index when remove_index targets an alias', () => {
+        const task = updateAliases(client, [
+          {
+            remove_index: {
+              index: 'existing_index_2_alias',
+            },
+          },
+        ]);
+        return expect(task()).resolves.toMatchInlineSnapshot(`
+                  Object {
+                    "_tag": "Left",
+                    "left": Object {
+                      "type": "remove_index_not_a_concrete_index",
+                    },
+                  }
+                `);
+      });
+    });
+  });
+
+  describe('createIndex', () => {
+    afterAll(async () => {
+      await client.indices.delete({ index: 'yellow_then_green_index' });
+    });
+    it('returns right after waiting for an index status to be green if the index already existed', async () => {
+      // Create a yellow index
+      await client.indices.create(
+        {
+          index: 'yellow_then_green_index',
+          body: {
+            mappings: { properties: {} },
+            settings: {
+              // Allocate 1 replica so that this index stays yellow
+              number_of_replicas: '1',
+            },
+          },
+        },
+        { maxRetries: 0 /** handle retry ourselves for now */ }
+      );
+
+      // Call createIndex even though the index already exists
+      const createIndexPromise = createIndex(client, 'yellow_then_green_index', undefined as any)();
+      let indexGreen = false;
+
+      setTimeout(() => {
+        client.indices.putSettings({
+          body: {
+            index: {
+              number_of_replicas: 0,
+            },
+          },
+        });
+        indexGreen = true;
+      }, 10);
+
+      return createIndexPromise.then((res) => {
```

Comment:
could you add negative scenarios (left) as well?

Reply:
there are no lefts for createIndex. I added a test that rejects because of an ES error so it basically asserts that we don't swallow rejections from the ES client.